### PR TITLE
Improving and fixes:

### PR DESCRIPTION
**Pull request by zloidemon**
- Added depedencies at setup.py (Markdown and crcmod)
- Replaced naming files from num by counting to generated crc hash by source
- Removing prefixes of plugin name
- Fix make png at PlantUML from tmp file
- Added more examples of usage plugins
- Changed checking <tag> with regexp because it did not work with jinja
- Removed extract_format function, extracting format from regexp group
- Improving speed with checking exist generated files

### CODE REVIEW

File: examples/example.py [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
+    extension  = sys.argv[1]
+    if extension in extensions:
+        fin = open("%s.md" % extension, 'r')
+        txt = fin.read()
+        fin.close()
+        md = markdown.Markdown(extensions=[extension])
+        print md.convert(txt)
+    else:
+        print "Extension(%s) do not exist" % extension
+except:
```
Comment (renstrom) [inline examples/example.py]: Even though this is just example code, we probably shouldn't use a bare except (see quote below).

```diff
-except:
+except Exception:
```

>A bare except: clause will catch SystemExit and KeyboardInterrupt exceptions, making it harder to interrupt a program with Control-C, and can disguise other problems. If you want to catch all exceptions that signal program errors, use except Exception: (bare except is equivalent to except BaseException: )

File: mdx_graphviz.py [hunk 2]
```diff
@@ -99,28 +102,34 @@ def run(self, lines):
...
+                                 self.graphviz.config["FORMAT"])
+            p = subprocess.Popen(cmd, shell=True,
+                                 stdin=subprocess.PIPE,
+                                 stdout=subprocess.PIPE, close_fds=True)
+            p.stdin.write(code)
+            p.stdin.close()
+            p.wait()
+            fout = open(filepath, 'w')
+            fout.write(p.stdout.read())
+            fout.close()
```
Comment (renstrom) [inline mdx_graphviz.py]: I realise it's copied from above, but change this to use `with` instead:

```python
with open(filepath, 'w') as f:
    f.write(p.stdout.read())
```

**Comment by vi4m** [comment]
What about it?

**Comment by renstrom** [comment]
Waiting for replies and/or fixes for my comments.

**Pull request merged**